presto-verifier/src/main/java/com/facebook/presto/verifier/rewrite/QueryRewriter.java

```diff
@@ -395,4 +430,271 @@ private static List<Property> applyPropertyOverride(List<Property> properties, L
+                .put(QualifiedName.of("approx_percentile"), new FunctionCallSubstitute(QualifiedName.of("avg"), ImmutableList.of(1)))
+                .put(QualifiedName.of("approx_distinct"), new FunctionCallSubstitute(QualifiedName.of("count"), ImmutableList.of(1)))
+                .build();
+
+        @Override
+        protected Node visitExpression(Expression node, Void context)
+        {
+            return ExpressionTreeRewriter.rewriteWith(new ExpressionRewriter<Void>()
+            {
+                @Override
+                public Expression rewriteFunctionCall(FunctionCall node, Void context, ExpressionTreeRewriter<Void> treeRewriter)
+                {
+                    FunctionCall defaultRewrite = treeRewriter.defaultRewrite(node, context);
+
+                    if (!FUNCTION_CALL_SUBSTITUTE_LOOKUP.containsKey(node.getName())) {
+                        return defaultRewrite;
+                    }
+
+                    FunctionCallSubstitute substitute = FUNCTION_CALL_SUBSTITUTE_LOOKUP.get(node.getName());
+                    QualifiedName rewrittenName = substitute.name;
+                    List<Expression> rewrittenArguments = substitute.originalArgumentIndices.stream()
+                            .map(originalIndex -> {
+                                Expression originalArgument = node.getArguments().get(originalIndex - 1);
+                                return treeRewriter.rewrite(originalArgument, context);
+                            }).collect(toImmutableList());
+
+                    return new FunctionCall(rewrittenName, defaultRewrite.getWindow(), defaultRewrite.getFilter(), defaultRewrite.getOrderBy(),
+                            defaultRewrite.isDistinct(), defaultRewrite.isIgnoreNulls(), rewrittenArguments);
+                }
+            }, node);
+        }
+
+        @Override
+        protected Node visitQuery(Query node, Void context)
+        {
+            return new Query(
+                    node.getWith().map(with -> (With) process(with, context)),
+                    (QueryBody) process(node.getQueryBody(), context),
+                    node.getOrderBy().map(orderBy -> (OrderBy) process(node, context)),
+                    node.getOffset(),
+                    node.getLimit());
+        }
+
+        @Override
+        protected Node visitWith(With node, Void context)
+        {
+            return new With(
+                    node.isRecursive(),
+                    node.getQueries().stream()
+                            .map(withQuery -> (WithQuery) process(withQuery, context))
+                            .collect(toImmutableList()));
+        }
+
+        @Override
+        protected Node visitWithQuery(WithQuery node, Void context)
+        {
+            return new WithQuery(node.getName(), (Query) process(node.getQuery(), context), node.getColumnNames());
+        }
```

Comment:
I'm not familiar with how an AstVisitor supposed to work, but would not it be enough to just override the methods that actually can modify the AST?
Like visitExpression() or visitExpressionItem()?
Overriding a bunch of methods that seem to construct the same nodes looks weird: it is like I want to change a single node type in a visitor and I need to override ALL visitor methods?